app/lib/search/top_packages.dart

```diff
@@ -140,7 +140,10 @@ CachedValue<List<PackageView>> _cachedValue(
         // Do not apply rate limit here.
         rateLimitKey: null,
       );
-      return page.packageHits;
+      // Returning `null` will not cache the value, returning empty list gets cached.
+      return page.errorMessage == null && page.packageHits.isNotEmpty
```

Comment:
Is it not enough to just look at the `page.errorMessage`?

Reply:
In theory (and with the current state of the code) it would be enough to just look for the `errorMessage`. On the other hand I think it wouldn't hurt to be more defensive and cache only non-empty results.